### ISSUE TITLE
[Feature request] ESPnet2 standalone Transducer

### ISSUE THREAD

**Issue by b-flo** [Masao-Someki/espnet_onnx#44]
Hi Masao!

I was wondering if you could consider adding support for the standalone version of ESPnet2 Transducer? See [the doc](https://espnet.github.io/espnet/espnet2_tutorial.html#transducer-asr).
I'm quite interested in ONNX but I have too much at hands right now to work on this project... I can help on the code understanding or similar things though.

**Comment by Masao-Someki** [comment]
Hi @b-flo, thank you!
Am I right that the new version of the transducer is implemented in `espnet2/asr_transducer`?
Also, I will try to support streaming inference with transducer models with that version!

**Comment by joazoa** [comment]
@Masao-Someki  yes, the standalone version is implemented there.  (I am using it)

**Comment by b-flo** [comment]
> Am I right that the new version of the transducer is implemented in espnet2/asr_transducer?

These files implement the standalone version:

- `espnet2/asr_transducer/*`
- `espnet2/tasks/asr_transducer.py`
- `espnet2/bin/asr_transducer_{inference,train}.py`

Dependency with ESPnet1 is minimal, only core modules related to ASR tasks and training utilities are used here.

> Also, I will try to support streaming inference with transducer models with that version!

Note that we rely on a unified online/offline architecture compared to other models depending on offline + contextual block architectures!

Also, some warning: there are a lot of pending/in-work things for this version so it may change quickly. Looking at my TO-DO though, there is not many items that would impact your development.

> yes, the standalone version is implemented there. (I am using it)

@joazoa Thanks for using it! If you've any feedback or report, feel free to post them on ESPnet! Any experiments outside our internal ones is appreciated right now!
